community/index/src/main/java/org/neo4j/index/gbptree/GBPTree.java

```diff
@@ -288,13 +288,21 @@ public GBPTree( PageCache pageCache, File indexFile, Layout<KEY,VALUE> layout, i
         this.freeList = new FreeListIdProvider( pagedFile, pageSize, rootId, FreeListIdProvider.NO_MONITOR );
         this.writer = new SingleIndexWriter( new InternalTreeLogic<>( freeList, bTreeNode, layout ) );
 
-        if ( created )
+        try
         {
-            initializeAfterCreation( layout );
+            if ( created )
+            {
+                initializeAfterCreation( layout );
+            }
+            else
+            {
+                loadState( pagedFile );
+            }
         }
-        else
+        catch ( Throwable e )
         {
-            loadState( pagedFile );
+            close();
+            throw e;
```

Comment:
If `close()` throws, for whatever reason, then that exception should be suppressed instead of shadowing the original exception.